### PR TITLE
chore(*) enforce use of correct errors module

### PR DESCRIPTION
**Pull request by jpeach**
### Summary

There are a number of Go modules named `errors` and it is easy
for autocompletion to pick the wrong on, since they are all
similar. Remove `github.com/go-errors/errors`, since Kuma has
already standardized on `github.com/pkg/errors`. Add a linter
rule to prevent regressions.

### Full changelog

N/A

### Issues resolved

N/A

### Documentation

N/A

### Testing

- [x] Unit tests
- [x] E2E tests
- [ ] Manual testing on Universal
- [ ] Manual testing on Kubernetes 

### Backwards compatibility

- [x] Add `backport-to-stable` label if the code is backwards compatible. Otherwise, list breaking changes.

### CODE REVIEW

**Review by lahabana — APPROVED** [review]

File: .golangci.yml [hunk 1]
```diff
@@ -33,6 +34,12 @@ linters-settings:
       alias: mesh_proto
     - pkg: github.com/kumahq/kuma/pkg/util/proto
       alias: util_proto
+  gomodguard:
+    blocked:
+      modules:
+      - github.com/go-errors/errors:
```
Comment (lobkovilya) [inline .golangci.yml:40]: what about just `errors` module?

Comment (jpeach) [reply .golangci.yml:40]: We could, if there's consensus. Though it would prevent adopting APIs like `errors.Is`.

Comment (jakubdyszkiewicz) [reply .golangci.yml:40]: wait, do we want to block the standard `errors` module? why?

**Review by jakubdyszkiewicz — APPROVED** [review]

File: .golangci.yml (hunk 1 above)
Comment (jpeach) [reply .golangci.yml:40]: > wait, do we want to block the standard `errors` module? why?

It is not blocked in this PR. The question was whether we ought to block it.

Comment (jakubdyszkiewicz) [reply .golangci.yml:40]: why would we block it? We use `errors.New` all the time

Comment (lobkovilya) [reply .golangci.yml:40]: `errors.New` is a function from `pkg/errors`

Comment (jpeach) [reply .golangci.yml:40]: Yeh, we use both versions.

**Pull request merged**